deepmd/fit/ener.py

```diff
@@ -276,8 +276,8 @@ def _build_lower(
     ):
         # cut-out inputs
         inputs_i = tf.slice (inputs,
-                             [ 0, start_index*      self.dim_descrpt],
-                             [-1, natoms* self.dim_descrpt] )
+                             [ 0, start_index, 0],
+                             [-1, natoms, self.dim_descrpt] )
```

Comment:
I think casting `natoms` from `int32` to `int64` will also work if you don't want to change dims.